addons/payment_fix_register_token/models/account_payment_register.py

```diff
@@ -69,3 +69,12 @@ def _create_payment_vals_from_wizard(self):
         payment_vals = super()._create_payment_vals_from_wizard()
         payment_vals['payment_token_id'] = self.payment_token_id.id
         return payment_vals
+
+    def _create_payments(self):
+        payments = super()._create_payments()
+        batches = self._get_batches()
+        # On register payment wizard, it only allow selecting payment token
+        # if processing single invoice and grouping invoices
+        if self.group_payment or len(batches[0]["lines"]) == 1:
+            payments.payment_transaction_id.write({"invoice_ids": [(6, 0, batches[0]["lines"].move_id.ids)]})
+        return payments
```

Comment:
Here is my suggestion:
```
    def _create_payments(self):
        # OVERRIDE
        payments = super()._create_payments()
        for pay in payments:
            if not pay.payment_transaction_id:
                continue

            reconciled_invoices = pay._get_reconciled_invoices()
            if reconciled_invoices:
                pay.payment_transaction_id.invoice_ids = reconciled_invoices
        return payments
```

Reply:
Indeed, I did try to use the reconciled invoices when doing the payment but such information does not exist during the transaction